snapcraft/extensions/kde_neon_qt6.py

```diff
@@ -0,0 +1,348 @@
+
+        return KDESnapsQt6(
+            qt6_sdk_snap=qt6_sdk_snap,
+            content_qt6=content_qt6_snap,
+            qt6_builtin=qt6_builtin,
+            gpu_layouts=gpu_layouts,
+            gpu_plugs=gpu_plugs,
+        )
+
+    @overrides
+    def get_root_snippet(self) -> Dict[str, Any]:
+        platform_qt6_snap = self.kde_snaps.content_qt6
+        content_qt6_snap = self.kde_snaps.content_qt6 + "-all"
+        gpu_plugs = self.kde_snaps.gpu_plugs
+        gpu_layouts = self.kde_snaps.gpu_layouts
+
+        return {
+            "assumes": ["snapd2.58.3"],  # for cups support
+            "compression": "lzo",
+            "plugs": {
+                "desktop": {"mount-host-font-cache": False},
+                "gtk-2-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/themes",
+                    "default-provider": "gtk-common-themes",
+                },
+                "gtk-3-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/themes",
+                    "default-provider": "gtk-common-themes",
+                },
+                "icon-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/icons",
+                    "default-provider": "gtk-common-themes",
+                },
+                "sound-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/sounds",
+                    "default-provider": "gtk-common-themes",
+                },
+                platform_qt6_snap: {
+                    "content": content_qt6_snap,
+                    "interface": "content",
+                    "default-provider": platform_qt6_snap,
+                    "target": "$SNAP/qt6",
+                },
+                **gpu_plugs,
+            },
+            "environment": {
+                "SNAP_DESKTOP_RUNTIME": "$SNAP/qt6",
+                "GTK_USE_PORTAL": "1",
+            },
+            "hooks": {
+                "configure": {
+                    "plugs": ["desktop"],
+                    "command-chain": ["snap/command-chain/hooks-configure-desktop"],
+                }
+            },
+            "layout": {
+                "/usr/share/X11": {"symlink": "$SNAP/qt6/usr/share/X11"},
+                "/usr/share/qt6": {"symlink": "$SNAP/qt6/usr/share/qt6"},
+                **gpu_layouts,
+            },
+        }
+
+    @overrides
+    def get_part_snippet(self, *, plugin_name: str) -> Dict[str, Any]:
+        qt6_sdk_snap = self.kde_snaps.qt6_sdk_snap
+
+        return {
+            "build-environment": [
+                {
+                    "PATH": prepend_to_env(
+                        "PATH",
+                        [
+                            "$CRAFT_STAGE/usr/bin",
+                            f"/snap/{qt6_sdk_snap}/current/usr/bin",
+                            "/usr/bin",
```

Comment:
I get the first two (prefer staged items to items in the SDK snap and include the SDK snap's bins on the path), but why are we prepending `/usr/bin` to the path here?